File: code/community/1304596979887640578/503-504-errors-deepgram-api.go
package main

import (
    "bytes"
    "encoding/json"
    "fmt"
    "net/http"
    "os"
)

func main() {
    deepgramApiKey := os.Getenv("DEEPGRAM_API_KEY")
    projectId := os.Getenv("PROJECT_ID")

    url := "https://api.deepgram.com/v1/listen"
    audioUrl := "https://example.com/audiofile.mp3"

    requestBody, _ := json.Marshal(map[string]string{
        "url":     audioUrl,
        "project": projectId,
    })

    req, err := http.NewRequest("POST", url, bytes.NewBuffer(requestBody))
    if err != nil {
        fmt.Println("Error creating request:", err)
        return
    }

    req.Header.Set("Authorization", "Token "+deepgramApiKey)
    req.Header.Set("Content-Type", "application/json")

    client := &http.Client{}
    resp, err := client.Do(req)
    if err != nil {
        fmt.Println("Error making HTTP request:", err)
        return
    }
    defer resp.Body.Close()

    if resp.StatusCode == http.StatusOK {
        var result map[string]interface{}
        json.NewDecoder(resp.Body).Decode(&result)
        fmt.Println("Transcription:", result)
    } else {
        fmt.Printf("Error: %s\n", resp.Status)
    }
}

File: code/community/1304596979887640578/503-504-errors-deepgram-api.rs
use std::env;
use deepgram::Deepgram;
use reqwest::Client;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let deepgram_api_key = env::var("DEEPGRAM_API_KEY").expect("DEEPGRAM_API_KEY not set");
    let project_id = env::var("PROJECT_ID").expect("PROJECT_ID not set");

    let client = Client::new();
    let deepgram = Deepgram::new(client, deepgram_api_key);
    
    let url = "https://api.deepgram.com/v1/listen";
    let audio_url = "https://example.com/audiofile.mp3";

    let response = deepgram.transcribe_audio_url(url, audio_url, &project_id).await?;
    println!("Transcription: {:?}", response);
    
    Ok(())
}